nemo/collections/llm/evaluation/base.py

```diff
@@ -74,7 +76,7 @@ def tokenizer_type(self, tokenizer):
         elif isinstance(tokenizer, SentencePieceTokenizer):
             return "SentencePieceTokenizer"
         else:
-            raise ValueError(
+            warnings.warn(
                 "Tokenizer type is not one of SentencePieceTokenizer or HF's AutoTokenizer. Please check "
                 "how to handle special tokens for this tokenizer"
             )
```

Comment:
Why is a warning being logged instead of raising an error now?

Reply:
The tokenizer type is needed here https://github.com/NVIDIA/NeMo/blob/c62cb7d1dc1a23aed47010916a1a6b5e8af14643/nemo/collections/llm/evaluation/base.py#L91-L96 only to decide special kwargs. In case of other tokenizers, we assume that no special kwargs are required unless explicitly added.

Reply:
Cool I can add this back.